src/effects/VST3/VST3Effect.cpp

```diff
@@ -339,6 +339,68 @@ bool VST3Effect::SupportsAutomation() const
    return false;
 }
 
+
+bool VST3Effect::FetchSettings(VST3EffectSettings& settings) const
+{
+   // partly copied from ::SaveSettings + some adaptations made
+
+   if (mEditController == nullptr)
+      return false;
+
+   using namespace Steinberg;
+
+   auto theCount = mEditController->getParameterCount();
+
+   for (int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
+   {
+      Vst::ParameterInfo parameterInfo{ };
+      if (mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+      {
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
+         {
+
+            Steinberg::Vst::ParamID    key = parameterInfo.id;
+            Steinberg::Vst::ParamValue value = mEditController->getParamNormalized(parameterInfo.id);
+
+            settings.mValues[key] = value;
+         }
+      }
+   }
+
+   return true;
+}
+
+
+bool VST3Effect::StoreSettings(const VST3EffectSettings& settings) const
+{
+   // partly copied from ::LoadSettings + some adaptations made
+
+   using namespace Steinberg;
+
+   if (mComponentHandler == nullptr)
+      return false;
+
+   for (const auto& item : settings.mValues)
```

Comment:
These functions to fetch settings from the VST3 handle into our own map, and store them back, are a beginning toward statelessness by analogy with the transformatio I am making for AudioUnits.

For AudioUnits, the fetching and storing both perform a loop over the settings as defined by the handle.  StoreSettings doesn't iterate our own hash map, but looks up in it.

Also in AudioUnits, the details of how to iterate the handle's metadata for settings is written just once in a higher-order function called `ForEachParameter`, and that function takes a lambda for the loop body.

I suggest you make similar changes.